### PR TITLE
Improve the user experience for view all disruptions page

### PR DESCRIPTION
**Pull request by alvak1**
Currently the view all disruptions page takes time to load and this PR fixes the issue to improve the user experience

### CODE REVIEW

File: site/components/map/ServicesMap.tsx [hunk 1]
```diff
@@ -287,22 +299,27 @@ const Map = ({
                     ),
                 );
 
+                const stops = includeMarkerData
+                    ? [...(state.inputs.stops ?? []), ...markerDataInAService, ...(searched.length > 0 ? searched : [])]
+                          .filter(
+                              (value, index, self) => index === self.findIndex((s) => s.atcoCode === value.atcoCode),
+                          )
+                          .splice(0, 100)
```
Comment (alainajaffer) [inline site/components/map/ServicesMap.tsx:307]: why are the stops changes showing here?

File: site/pages/api/get-all-disruptions.test.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    const getDisruptionsDataFromDynamoSpy = vi.spyOn(dynamo, "getDisruptionsDataFromDynamo");
+    const sortDisruptionsByStartDateSpy = vi.spyOn(utils, "sortDisruptionsByStartDate");
+
+    afterEach(() => {
+        vi.resetAllMocks();
+    });
+
+    // beforeEach(() => {
+    //     getDisruptionsDataFromDynamoSpy.mockImplementation(() => {
+    //         return disruptions;
```
Comment (alainajaffer) [inline site/pages/api/get-all-disruptions.test.ts:26]: remove commented code

File: site/pages/view-all-disruptions.page.tsx [hunk 3]
```diff
@@ -353,15 +348,26 @@ const applyFiltersToDisruptions = (
...
+export const getDisruptionData = async () => {
+    const options: RequestInit = {
+        method: "GET",
+        headers: {
+            "Content-Type": "application/json",
+        },
+    };
+
+    const res = await fetch("/api/get-all-disruptions", options);
+    const disruptions = (await res.json()) as TableDisruption[];
```
Comment (alainajaffer) [inline site/pages/view-all-disruptions.page.tsx:360]: could we use zod here to parse it?

File: site/pages/view-all-disruptions.test.tsx [hunk 4]
```diff
@@ -68,87 +78,114 @@ const disruptions: TableDisruption[] = [
...
         });
 
-        it("should render correctly when filter is set to pending approval status", () => {
-            useRouter.mockImplementation(() => ({
-                query: { pending: true },
-            }));
+        it("should render correctly when filter is set to pending approval status", async () => {
+            fetchSpy.mockResolvedValue({
+                json: vi.fn().mockResolvedValue([...disruptions, ...disruptions, ...disruptions, ...disruptions]),
+            } as unknown as Response);
```
Comment (alainajaffer) [inline site/pages/view-all-disruptions.test.tsx:154]: why as unknown as Response?

File: site/pages/view-all-disruptions.test.tsx [hunk 5]
```diff
@@ -68,87 +78,114 @@ const disruptions: TableDisruption[] = [
...
+const fetchSpy = vi.spyOn(global, "fetch");
+
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const useRouter = vi.spyOn(require("next/router"), "useRouter");
 beforeEach(() => {
     useRouter.mockImplementation(() => ({
         query: "",
     }));
+
+    vi.spyOn(global, "fetch");
```
Comment (alainajaffer) [inline site/pages/view-all-disruptions.test.tsx:90]: is this needed if we have fetchSpy above?

File: site/pages/view-all-disruptions.test.tsx (hunk 4 above)
Comment (alainajaffer) [reply site/pages/view-all-disruptions.test.tsx:154]: As discussed

File: site/pages/view-all-disruptions.test.tsx (hunk 5 above)
Comment (DeannaSharma) [reply site/pages/view-all-disruptions.test.tsx]: resolved

File: site/pages/view-all-disruptions.page.tsx (hunk 3 above)
Comment (DeannaSharma) [reply site/pages/view-all-disruptions.page.tsx]: resolved

**Review by alainajaffer — APPROVED** [review]

**Pull request merged**